pyodide-build/pyodide_build/create_xbuildenv.py

```diff
@@ -29,12 +29,51 @@ def copy_xbuild_files(xbuildenv_path):
             shutil.copy(site_packages / path, target)
 
 
+def get_relative_path(pyodide_root, flag):
+    return Path(get_make_flag(flag)).relative_to(pyodide_root)
+
+
 def main(args):
     pyodide_root = get_pyodide_root()
+    pythoninclude = get_relative_path(pyodide_root, "PYTHONINCLUDE")
+    wasm_lib_dir = get_relative_path(pyodide_root, "WASM_LIBRARY_DIR")
+    sysconfig_dir = get_relative_path(pyodide_root, "SYSCONFIGDATA_DIR")
     xbuildenv_path = pyodide_root / "xbuildenv"
     shutil.rmtree(xbuildenv_path, ignore_errors=True)
+
+    xbuildenv_root = xbuildenv_path / "pyodide-root"
     xbuildenv_path.mkdir()
-    shutil.copytree(get_make_flag("PYTHONINCLUDE"), xbuildenv_path / "python-include")
+    to_copy: list[Path] = [
+        pythoninclude,
+        sysconfig_dir,
+        Path("Makefile.envs"),
+        wasm_lib_dir / "CLAPACK",
+        wasm_lib_dir / "cmake",
+        Path("tools/pyo3_config.ini"),
+    ]
+    # Some ad-hoc stuff here to moderate size. We'd like to include all of
+    # wasm_lib_dir but there's 180mb of it. Better to leave out all the video
+    # codecs and stuff.
+    for pkg in ["ssl", "libcrypto", "zlib", "xml", "mpfr"]:
```

Comment:
So this assumes that all of libraries are already built and we copy them into a isolated xbuild env?

Reply:
If you're curious, an example of a working xbuildenv is here:
http://pyodide-cache.s3-website-us-east-1.amazonaws.com/xbuildenv/1.tar
If you install `pyodide-build` from my `out-of-tree` branch:
```
pip install git+https://github.com/hoodmane/pyodide.git@out-of-tree#subdirectory=pyodide-build
```
then you can build e.g., numpy with:
```
pywasmbuild
```